update stage names [SWAT-1234]

morbalint opened the pull request.

morbalint committed:
update stage names [SWAT-1234]

DamienBitrise commented on recipes/ios-run-test-groups-in-parallel.md:
```diff
@@ -48,21 +48,21 @@ meta:
 pipelines:
   run_tests_groups:
```
this is the same name as the stage

morbalint replied:
renamed the pipeline to just `run_tests` (to avoid updating the screenshot)

DamienBitrise commented on recipes/ios-run-test-groups-in-parallel.md:
```diff
@@ -48,21 +48,21 @@ meta:
     stages:
-    - build_tests: {}
+    - build_tests_stage: {}
     - run_tests_groups: {}
```
this is the same name as the pipeline

DamienBitrise replied:
oh sorry misread the code looks good

morbalint committed:
rename pipline to `run_tests`

to avoid using the same name for the pipeline and the stage so that it's easier to understand

DamienBitrise reviewed (approved):
LGTM

godrei commented:
I miss consistency in the naming of the entities:
- (stage) `build_for_ui_testing` -> `build`: I feel like the `build` name is a bit vague
- I feel like we should be consistent with the naming, it looks strange that some entity has their type in their name, but not in every entity with the same type and across types
  - (stage) `build_tests` -> `build_tests_stage`
  - (stage) `deploy_test_results` -> `deploy_test_results_stage`
  - (workflow) `build_tests` -> `build_tests_workflow`
  - (workflow) `deploy_test_results` -> `deploy_test_results_workflow`

---


**What about using the following principles?**

- Entity names should be unique (both within and between types) -> to resolve the confusion when you read or type the `artifact_sources` (`<stage>.<workflow>`)
- As a rule of thumb: the length of entity names increases towards lower-level entities (pipeline -> workflow) -> as you approach lower-level entities, the names will tell more details
    - Pipeline is the top-level entity in the bitrise.yml, it's name is less important because you will probably have the least pipeline (compared to stages and workflows), it can be as simple as `ci`, `primary`, `deploy`, `test`, ... or can be constructed based on the stage names, like `build_and_run_tests`

**With the above principles, this is how the entity names would look in the examples:**

_(iOS) Run tests in parallel on multiple simulators_

```
build_and_run_tests
- build_tests
    xcode_build_for_test
- run_tests # artifact_sources: build_tests.xcode_build_for_test
    run_tests_on_iPhone
    run_tests_on_iPad
    run_tests_on_iPod
```

_(iOS) Run test groups in parallel_

```
build_and_run_tests
- build_tests
    xcode_build_for_test
- run_test # artifact_sources: build_tests.xcode_build_for_test
    run_ui_tests
    run_unit_tests
```

_(iOS) Merging test results and deploying to the Test Reports add-on_

```
build_and_run_tests
- build_tests
    xcode_build_for_test
- run_tests # artifact_sources: build_tests.xcode_build_for_test
    run_ui_tests
    run_unit_tests
- deploy_test_results # artifact_sources: run_tests\..*
    merge_and_deploy_test_results
```

morbalint committed:
address review comments

morbalint committed:
remove _group from stage name

morbalint committed:
rename stages to conform with guidelines
